src/v/cluster/cluster_discovery.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+
+#include <optional>
+#include <vector>
+
+namespace cluster {
+
+// Provides metadata pertaining to initial cluster discovery.
+class cluster_discovery {
+public:
+    explicit cluster_discovery(const model::node_uuid& node_uuid);
+
+    // Returns this node's node ID.
+    //
+    // TODO: implement the below behavior.
+    //
+    // Determines what the node ID for this node should be. Once called, we can
+    // proceed with initializing anything that depends on node ID.
+    //
+    // On an empty seed server, this sends requests to all other seed servers
+    // to determine if there is a valid assigned of node IDs for the seeds.
+    //
+    // On a non-seed server with no node ID specified via config, this sends a
```

Comment:
could you clarify the difference between:

1. `empty seed server`
2. `non-seed server`

i think i know, but the terminology seems confusing

Reply:
Sure, an empty seed server is a seed server that has no data on it. A non-seed server is a server that isn't in the seed server list. I updated this comment a bit to clarify "empty seed server".

Reply:
oh i see. 'empty seed server' was confusing because 'seed server' had been identified as 'server with an empty seed server list' and now it is 'a seed server (empty seed server list) whose directory is also empty`? anyway, thanks for updating the comment.

Reply:
Ah, yeah I see how that's easy to misinterpret. Glad it's clearer now